pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1051,6 +1064,13 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulat
 				return nil, err
 			}
 			logger.Info("Domain defined.")
+			// this is the point where we need to build the devices metadata if it was requested.
+			// This metadata maps the user provided tag to the hypervisor assigned device address.
+			err = buildDevicesMetadata(vmi, dom)
```

Comment:
is there a way we can avoid generating the cloud-init iso twice when tags are used?

maybe we need a `postDefineHook(vmi, domain)` that contains the items in preStartHook that need to occur after we know more info about the domain

Reply:
all interfaces which can have tags are already present on the VMI, I don't think that we need to take hooks into account at all.

Reply:
ah of course we need to think about the hook, to take e.g. pci address changes from the hook into account. Sorry.

Reply:
good idea. Let me check if we can define the domain without having the file on disk.